### ISSUE TITLE
Examples

### ISSUE THREAD

**Issue by eldad87** [megalepozy/grpcweberr#1]
Hi,
Is there any chance you can add some examples?
It's hard to understand how to utilize your library just by reading your code.

Thanks!

**Comment by Megalepozy** [comment]
Hey Eldad,

Yeah I know and totally understand what u mean, I'll do it now :)

**Comment by Megalepozy** [comment]
OK it's done, let me know if something is unclear :)

**Comment by eldad87** [comment]
Much better!
IMO, you should extent your library to support validation error(s).
So you could set an error message like you do now, but in case and the error is a result of an invalid argument(s) - I would like to share that info with my Client.

For example:
```
err := grpcweberr.New("lcaxafdv23d9d2s", codes.InvalidArgument, 422, "Received invalid values")
err.withDetails("username", "Username already exists")
```

WDYT?

**Comment by Megalepozy** [comment]
I totally agree with u about the need to support multiple errors, for instance when a user fill a form with multiple fields, I thought about it but since I don't really have a need for it at my current project it never got implemented... :)

Currently I use the msg parameter for the client err message when its a user error (for instance "Username already exists") and when its a system error I set it to "" so it will get the default ambiguous error. 

Honestly, it's far from being elegant, what may be a nice solution is to remove the msg parameter entirely and change the func signature to variadic function signature which accept functions as described at https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis (I implemented it already at [sdlog](https://github.com/Megalepozy/sdlog) and really liked it) so the new signature will be along the lines of:
func New(id string, code codes.Code, http int, msgs ...func(e *error)) error 
This way New could be called with unlimited "msg functions" to add unlimited # of messages while also allowing to just return the default msg when no msg func is supplied.

Other then that I would love to remove the id param as well (so at minimum supplying 2 values will be enough for New()) since the user shouldn't care about such a thing and when its a system error I anyway log its location and stack trace... so its kinda useless at least for me.

Lastly, I need to add versioning since maybe someone will actually use this pkg and I don't want to break functionality like that :)

Let me know what u think and if u need such changes (would love to know if u intend to use this pkg :)), if so I'll try to do it in the next few days, but feel free to make a pull request! ;)
